pipeline/steps_test.go

```diff
@@ -369,3 +372,13 @@ func Test_updateObjectStatus(t *testing.T) {
 	assert.Equal(t, "token-1234", updatedCluster.Status.BootstrapToken.Token)
 	assert.NotEqual(t, updatedCluster.ResourceVersion, cluster.GetResourceVersion())
 }
+
+func addLogger(c *Context) *Context {
+	l, err := zap.NewDevelopment()
+	if err != nil {
+		panic(fmt.Errorf("unexpected error while creating zap logger: %w", err))
```

Comment:
Not sure if I like the panic here. But I guess when logging to stdout this should never happen. So I guess its fine

Reply:
Would need some major test refactoring otherwisem, since contexts are created outside the test functions.